### PR TITLE
Update Makefile to not allow unset variables

### PR DESCRIPTION
**Pull request by s-fairchild**
Adding SHELLFLAGS and switch to double dollar signs to prevent make from expanding bash variables

### Which issue this PR addresses:

No current issues.

### What this PR does / why we need it:

Currently unset variables are allowed in Makefile. I don't believe there should be a case where we need/want unset variables, I believe this a safer option.  

`.SHELLFLAGS := -o nounset -c` sets `nounset` and variables are expanded by bash by escaping with double dollar signs.

### Test plan for issue:
Existing tests that utilize make

### Is there any documentation that needs to be updated for this PR?

No

### CODE REVIEW

**Review by facchettos — DISMISSED** [review]
:+1:

**Review by SrinivasAtmakuri — DISMISSED** [review]
LGTM!

**Review by dem4gus — DISMISSED** [review]
Good change 👍

File: Makefile [hunk 1]
```diff
@@ -132,7 +133,7 @@ publish-image-e2e: image-e2e
 	docker push $(ARO_IMAGE)
 
 extract-aro-docker:
-	hack/ci-utils/extractaro.sh ${ARO_IMAGE}
+	hack/ci-utils/extractaro.sh $${ARO_IMAGE:-}
```
Comment (cadenmarchese) [inline Makefile:136]: I think this is where your e2e failure is coming from:
```suggestion
	hack/ci-utils/extractaro.sh $${ARO_IMAGE}
```
~~~
+ DOCKERID=
make: *** [Makefile:136: extract-aro-docker] Error 1
~~~

Comment (s-fairchild) [reply Makefile]: I was able to fix the e2e pipeline by hardcoding `ARO_IMAGE` here: https://github.com/Azure/ARO-RP/blob/ce4e22ae71e5d2886db1cdba5857ef19e067a682/.pipelines/e2e.yml#L47

**Comment by cadenmarchese** [comment]
I also ran build and push pipeline to ensure that functionality isn't impacted:

https://msazure.visualstudio.com/AzureRedHatOpenShift/_build/results?buildId=71135590&view=results

**Pull request merged**